game.sy

```diff
@@ -238,6 +270,95 @@ update :: fn delta: float -> void {
         p.try_grab' puck
     }
 
+    iter := 1000
+    t := 1.0
+    loop t > 0.0 && iter > 0 {
+        iter -= 1
+        is_collision := false
+        next_col := collision.Manifold {
+            t: 2.0,
+            normal: (0.0, 0.0),
+            a: collision.Wall {
+                pos: (0.0, 0.0),
+                normal: (0.0, 0.0),
+            },
+            b: collision.Wall {
+                pos: (0.0, 0.0),
+                normal: (0.0, 0.0),
+            },
+        }
+
+        // Find next collision
+        a_i := 0
+        // TODO(ed): Write this with for_each, when the upvalue bug is fixed.
+        loop a_i < len(bodies) {
```

Comment:
This is a loop because of a language bug - I didn't manage to fix this one while I was at it. :(